### ISSUE TITLE
 Errors OpenAI as a representation model and 

### ISSUE THREAD

**Issue by mjaved-nz** [MaartenGr/BERTopic#1684]
Hi @MaartenGr

I am using bertopic==0.16.0 (with default settings) in a dedicated environment and getting the following errors:

Code:

                              # KeyBERT
                              keybert = KeyBERTInspired()
                              
                              # MMR
                              mmr = MaximalMarginalRelevance(diversity=0.3)
                              
                              representation_models = [mmr, rep_model_chatgpt]
                              
                              topic_model = BERTopic(language="english", 
						                              top_n_words=100,
                                                      verbose=True, 
                                                      seed_topic_list=seed_topic_list,
                                                      representation_model=representation_models,
                                                      vectorizer_model=CountVectorizer(ngram_range=(1, 3) , stop_words="english")
                                                      ) 
                              
                              
                              topics, probs = topic_model.fit_transform(docs)
                              topic_model.save('my_model')

1. When I use OpenAI as a representation model, it generate the following error when trying to save the model using model.save().

Error: 
2023-12-11 16:26:00,118 - BERTopic - WARNING: When you use `pickle` to save/load a BERTopic model,please make sure that the environments in which you saveand load the model are **exactly** the same. The version of BERTopic,its dependencies, and python need to remain the same.
Traceback (most recent call last):
…
TypeError: cannot pickle '_thread.RLock' object

2. When tried bertopic.fit_transform() with seed_topic_list for guided topic modelling (with or without rewpresentation model). getting the following error
Error:
 topics, probs = topic_model.fit_transform(docs)
  File "\.conda\envs\bertopic2\lib\site-packages\bertopic\_bertopic.py", line 399, in fit_transform
    y, embeddings = self._guided_topic_modeling(embeddings)
  File "\.conda\envs\bertopic2\lib\site-packages\bertopic\_bertopic.py", line 3617, in _guided_topic_modeling
    embeddings[indices] = np.average([embeddings[indices], seed_topic_embeddings[seed_topic]], weights=[3, 1])
  File "\.conda\envs\bertopic2\lib\site-packages\numpy\lib\function_base.py", line 511, in average
    a = np.asanyarray(a)
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.

Thanks

**Comment by MaartenGr** [comment]
@mjaved-nz I believe there is a known issue with pickle and OpenAI (check the issues) but you could resolve quite easily with this before saving:

```python
topic_model.representation_model = None
```

Other than that, saving with `safetensors` or `pytorch` will also work.

**Comment by mjaved-nz** [comment]
Thanks @MaartenGr ,

any suggestion about the following error

When tried bertopic.fit_transform() with seed_topic_list for guided topic modelling (with or without rewpresentation model). getting the following error
Error:
topics, probs = topic_model.fit_transform(docs)
File ".conda\envs\bertopic2\lib\site-packages\bertopic_bertopic.py", line 399, in fit_transform
y, embeddings = self._guided_topic_modeling(embeddings)
File ".conda\envs\bertopic2\lib\site-packages\bertopic_bertopic.py", line 3617, in _guided_topic_modeling
embeddings[indices] = np.average([embeddings[indices], seed_topic_embeddings[seed_topic]], weights=[3, 1])
File ".conda\envs\bertopic2\lib\site-packages\numpy\lib\function_base.py", line 511, in average
a = np.asanyarray(a)
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.

**Comment by MaartenGr** [comment]
@mjaved-nz I believe this is a known issue that has been raised before. I would suggest searching for this problem in both open and closed issues. I think you can find the solution there.

**Comment by mjaved-nz** [comment]
Hi @MaartenGr,

I am trying to merge some topics generated from the above model but getting the topic's names without representation. How can I use the representation_models for topic_model.merge_topics(docs, topics_to_merge)?

Thanks

**Comment by MaartenGr** [comment]
Using `.merge_topics` should create updated representations for each topic. To access them, either run `topioc_model.get_topic_info()` or run `topic_model.get_topics(full=True)`.

**Comment by mjaved-nz** [comment]
Thanks @MaartenGr,

I think my question was not clear. let me rephrase it. By using representation_models (gpt-3.5-turbo), I am getting topic's names as "12_Research team leading clinical trials with Biotech" in a meaningful way, but after merging topics getting as "12_trials_clinical_trial_clinical trials" as generated by the base model, not by the chatgpt.

**Comment by MaartenGr** [comment]
Sure, you can use `.update_topics` for that. Just make sure you specify the `representation_model`  parameter to use whatever representation you want.
